src/Mollie/WC/Helper/ApplePayDataObjectHttp.php

```diff
@@ -0,0 +1,470 @@
+    public $billingAddress;
+    /**
+     * @var string[]
+     */
+    public $shippingAddress;
+    /**
+     * @var mixed
+     */
+    public $callerPage;
+
+    /**
+     * @var array
+     */
+    public $errors = [];
+
+    /**
+     * Resets the errors array
+     */
+    protected function resetErrors()
+    {
+        $this->errors = [];
+    }
+
+    /**
+     * Returns if the object has any errors
+     * @return bool
+     */
+    public function hasErrors()
+    {
+        return !empty($this->errors);
+    }
+
+    /**
+     * Set the object with the data relevant to ApplePay validation
+     * @param array $post
+     */
+    public function validationData(array $post)
+    {
+        $this->resetErrors();
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            self::VALIDATION_REQUIRED_FIELDS
+        )
+        ) {
+            return;
+        }
+        $this->assignDataObjectValues($post);
+    }
+
+    /**
+     * Set the object with the data relevant to ApplePay on update shipping contact
+     * Required data depends on callerPage
+     * @param array $post
+     */
+    public function updateContactData(array $post)
+    {
+        $this->resetErrors();
+        $requiredFields = self::UPDATE_CONTACT_SINGLE_PROD_REQUIRED_FIELDS;
+        if (array_key_exists('callerPage', $post)
+            && $post['callerPage'] == 'cart'
+        ) {
+            $requiredFields = self::UPDATE_CONTACT_CART_REQUIRED_FIELDS;
+        }
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            $requiredFields
+        )
+        ) {
+            return;
+        }
+        $this->assignDataObjectValues($post);
+        $this->simplifiedContact = $this->simplifiedAddress(
+            filter_var_array(
+                $post['simplifiedContact'],
+                FILTER_SANITIZE_STRING
+            )
+        );
+    }
+
+    /**
+     * Set the object with the data relevant to ApplePay on update shipping method
+     * Required data depends on callerPage
+     *
+     * @param array $post
+     */
+    public function updateMethodData(array $post)
+    {
+        $this->resetErrors();
+        $requiredFields = self::UPDATE_METHOD_SINGLE_PROD_REQUIRED_FIELDS;
+
+        if (array_key_exists('callerPage', $post)
+            && $post['callerPage'] == 'cart'
+        ) {
+            $requiredFields = self::UPDATE_METHOD_CART_REQUIRED_FIELDS;
+        }
+        if (!$this->hasRequiredFieldsValues(
+            $post,
+            $requiredFields
+        )
+        ) {
+            return;
+        }
+        $this->assignDataObjectValues($post);
+
+        $this->simplifiedContact = $this->simplifiedAddress(
+            filter_var_array(
+                $post['simplifiedContact'],
+                FILTER_SANITIZE_STRING
+            )
+        );
+        $this->shippingMethod = filter_var_array(
+            $post['shippingMethod'],
+            FILTER_SANITIZE_STRING
+        );
+    }
```

Comment:
I have the impression that, part of this function has duplications with the previous one, things which can be avoided by producing an additional method for the common code.